test/ergw_pgw_test_lib.erl

```diff
@@ -776,72 +778,127 @@ validate_cause(_Type, {_, _, _} = SubType,
 validate_cause(_Type, _SubType, Response) ->
     ?match(#gtp{ie = #{{v2_cause,0} := #v2_cause{v2_cause = request_accepted}}}, Response).
 
+validate_seq_no(#gtp{seq_no = SeqNo}, #gtpc{seq_no = GtpCSeqNo}) ->
+    ?equal(GtpCSeqNo, SeqNo);
+validate_seq_no(#gtp{seq_no = SeqNo}, ExpectedSeqNo) ->
+    ?equal(ExpectedSeqNo, SeqNo).
+
+validate_teid(#gtp{tei = TEID}, #gtpc{local_control_tei = GtpCTEID}) ->
+    ?equal(GtpCTEID, TEID);
+validate_teid(#gtp{tei = TEID}, ExpectedTEID) ->
+    ?equal(ExpectedTEID, TEID).
+
 validate_response(_Type, system_failure, Response, GtpC) ->
+    validate_seq_no(Response, GtpC),
```

Comment:
Could these two (validate_seq_no, validate_teid) be done in one place, instead of in each validate_response/4 ?

Reply:
No, some procedures have special rules, (e.g. Seq-No with highest bit set, TEID == 0)